### DISCUSSION TITLE
Raspberry Pi 4 4gb not connecting after fresh OS install

### DISCUSSION THREAD

**Discussion by reillysmithers** [neutrinolabs/xrdp#2743] [Q&A]
I had previously setup XRDP with my Raspberry Pi 4 Running Bullseye 64 bit, I had set this up in early March and was therefore likely using the 22/09/22 RpiOS version. It was working very well, and I could remotely access my camera feed and other live views whilst using remote desktop. 
After an SD card failure requiring a re-flash of the OS, I'm trying to set up XRDP again but cannot connect whilst the user belongs to video and render groups, which are a necessity for using the camera, camera previews and other preview windows (for example openCV)
I've tried disabling autologin
I've tried various combinations of removing/adding user to video/render groups
I've tried the modifying xorg.conf and changing the "Option "DRMDevice"... line
I've tried both the 22/09/22 RpiOS version, and the latest versions
I've tried all versions of XRDP available from apt, and compiled from source too 
I've tried all these things in various combinations in attempt to restore functionality. None work.
Any and all help is appreciated, I have spent many days on this now without success, I get a black screen every time, despite having it previously working flawlessly.

**Comment by matt335672** [comment]
The DRMDevice mod [here](/neutrinolabs/xrdp/issues/2053#issuecomment-1038142308) is definitely the way to go.

I suggest something like the following:-
- Stick with the stock xrdp and xorgxrdp packages.
- Get the login working with the groups removed from the user.
- add the user to the groups and make the change to the DRMDevice.

If it's not working, have a look in `~/.xorgxrdp.*.log`. It's possible there is some other permission issue or hardware auto-adding going on that we haven't anticipated.

As always, be careful not to have the same user logged in on the console at the same time. That can confuse things dreadfully.

**Reply by reillysmithers** [reply]
Thanks for the response! As other users have found, the xorg.conf DRMDevice line modification works (I can login) but is very slow/unstable. I'll post xorgxrdp logs for you when I have time because something else is surely going wrong.

**Reply by reillysmithers** [reply]
X.Org X Server 1.20.11
X Protocol Version 11, Revision 0
[    29.865] Build Operating System: linux Debian
[    29.865] Current Operating System: Linux raspberrypi 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64
[    29.865] Kernel command line: coherent_pool=1M 8250.nr_uarts=0 snd_bcm2835.enable_headphones=0 snd_bcm2835.enable_headphones=1 snd_bcm2835.enable_hdmi=1 snd_bcm2835.enable_hdmi=0  smsc95xx.macaddr=DC:A6:32:52:21:0A vc_mem.mem_base=0x3ec00000 vc_mem.mem_size=0x40000000  console=ttyS0,115200 console=tty1 root=PARTUUID=cb124e50-02 rootfstype=ext4 fsck.repair=yes rootwait quiet splash plymouth.ignore-serial-consoles
[    29.865] Build Date: 30 March 2023  01:46:14PM
[    29.865] xorg-server 2:1.20.11-1+rpt3+deb11u6 (https://www.debian.org/support) 
[    29.865] Current version of pixman: 0.40.0
[    29.865] 	Before reporting problems, check http://wiki.x.org
	to make sure that you have the latest version.
[    29.865] Markers: (--) probed, (**) from config file, (==) default setting,
	(++) from command line, (!!) notice, (II) informational,
	(WW) warning, (EE) error, (NI) not implemented, (??) unknown.
[    29.865] (++) Log file: ".xorgxrdp.10.log", Time: Thu Jun 29 13:14:27 2023
[    29.867] (++) Using config file: "/etc/X11/xrdp/xorg.conf"
[    29.867] (==) Using system config directory "/usr/share/X11/xorg.conf.d"
[    29.868] (**) Option "defaultserverlayout" "X11 Server"
[    29.868] (**) ServerLayout "X11 Server"
[    29.868] (**) |-->Screen "Screen (xrdpdev)" (0)
[    29.868] (**) |   |-->Monitor "Monitor"
[    29.868] (**) |   |-->Device "Video Card (xrdpdev)"
[    29.868] (**) |-->Input Device "xrdpMouse"
[    29.868] (**) |-->Input Device "xrdpKeyboard"
[    29.868] (**) Option "DontVTSwitch" "on"
[    29.868] (**) Option "AutoAddDevices" "off"
[    29.868] (**) Not automatically adding devices
[    29.868] (==) Automatically enabling devices
[    29.868] (==) Automatically adding GPU devices
[    29.868] (==) Max clients allowed: 256, resource mask: 0x1fffff
[    29.869] (WW) The directory "/usr/share/fonts/X11/cyrillic" does not exist.
[    29.869] 	Entry deleted from font path.
[    29.869] (==) FontPath set to:
	/usr/share/fonts/X11/misc,
	/usr/share/fonts/X11/100dpi/:unscaled,
	/usr/share/fonts/X11/75dpi/:unscaled,
	/usr/share/fonts/X11/Type1,
	/usr/share/fonts/X11/100dpi,
	/usr/share/fonts/X11/75dpi,
	built-ins
[    29.869] (==) ModulePath set to "/usr/lib/xorg/modules"
[    29.869] (II) Loader magic: 0x555d38de28
[    29.869] (II) Module ABI versions:
[    29.869] 	X.Org ANSI C Emulation: 0.4
[    29.869] 	X.Org Video Driver: 24.1
[    29.869] 	X.Org XInput driver : 24.1
[    29.869] 	X.Org Server Extension : 10.0
[    29.874] (II) systemd-logind: took control of session /org/freedesktop/login1/session/c2
[    29.877] (II) xfree86: Adding drm device (/dev/dri/card1)
[    29.877] (EE) systemd-logind: failed to take device /dev/dri/card1: Operation not permitted
[    29.878] (EE) /dev/dri/card1: failed to set DRM interface version 1.4: Permission denied
[    29.878] (II) xfree86: Adding drm device (/dev/dri/card0)
[    29.879] (EE) systemd-logind: failed to take device /dev/dri/card0: Operation not permitted
[    29.880] (II) no primary bus or device found
[    29.880] 	falling back to /sys/devices/platform/v3dbus/fec00000.v3d/drm/card0
[    29.880] (II) "glx" will be loaded. This was enabled by default and also specified in the config file.
[    29.880] (II) LoadModule: "dbe"
[    29.881] (II) Module "dbe" already built-in
[    29.881] (II) LoadModule: "ddc"
[    29.881] (II) Module "ddc" already built-in
[    29.881] (II) LoadModule: "extmod"
[    29.881] (II) Module "extmod" already built-in
[    29.881] (II) LoadModule: "glx"
[    29.881] (II) Loading /usr/lib/xorg/modules/extensions/libglx.so
[    29.884] (II) Module glx: vendor="X.Org Foundation"
[    29.884] 	compiled for 1.20.11, module version = 1.0.0
[    29.884] 	ABI class: X.Org Server Extension, version 10.0
[    29.884] (II) LoadModule: "int10"
[    29.884] (II) Loading /usr/lib/xorg/modules/libint10.so
[    29.889] (II) Module int10: vendor="X.Org Foundation"
[    29.889] 	compiled for 1.20.11, module version = 1.0.0
[    29.889] 	ABI class: X.Org Video Driver, version 24.1
[    29.889] (II) LoadModule: "record"
[    29.889] (II) Module "record" already built-in
[    29.889] (II) LoadModule: "vbe"
[    29.890] (II) Loading /usr/lib/xorg/modules/libvbe.so
[    29.891] (II) Module vbe: vendor="X.Org Foundation"
[    29.891] 	compiled for 1.20.11, module version = 1.1.0
[    29.891] 	ABI class: X.Org Video Driver, version 24.1
[    29.891] (II) LoadModule: "glamoregl"
[    29.891] (II) Loading /usr/lib/xorg/modules/libglamoregl.so
[    29.900] (II) Module glamoregl: vendor="X.Org Foundation"
[    29.900] 	compiled for 1.20.11, module version = 1.0.1
[    29.900] 	ABI class: X.Org ANSI C Emulation, version 0.4
[    29.900] (II) LoadModule: "xorgxrdp"
[    29.901] (II) Loading /usr/lib/xorg/modules/libxorgxrdp.so
[    29.905] (II) Module XORGXRDP: vendor="X.Org Foundation"
[    29.905] 	compiled for 1.20.11, module version = 0.2.17
[    29.905] 	ABI class: X.Org Video Driver, version 24.1
[    29.905] xorgxrdpSetup:
[    29.905] (II) LoadModule: "fb"
[    29.905] (II) Loading /usr/lib/xorg/modules/libfb.so
[    29.906] (II) Module fb: vendor="X.Org Foundation"
[    29.906] 	compiled for 1.20.11, module version = 1.0.0
[    29.906] 	ABI class: X.Org ANSI C Emulation, version 0.4
[    29.906] (II) LoadModule: "xrdpdev"
[    29.906] (II) Loading /usr/lib/xorg/modules/drivers/xrdpdev_drv.so
[    29.908] (II) Module XRDPDEV: vendor="X.Org Foundation"
[    29.908] 	compiled for 1.20.11, module version = 0.2.17
[    29.908] 	ABI class: X.Org Video Driver, version 24.1
[    29.908] xrdpdevSetup:
[    29.908] (II) LoadModule: "xrdpmouse"
[    29.908] (II) Loading /usr/lib/xorg/modules/input/xrdpmouse_drv.so
[    29.909] (II) Module XRDPMOUSE: vendor="X.Org Foundation"
[    29.909] 	compiled for 1.20.11, module version = 0.2.17
[    29.909] 	Module class: X.Org XInput Driver
[    29.909] 	ABI class: X.Org XInput driver, version 24.1
[    29.909] rdpmousePlug:
[    29.909] (II) LoadModule: "xrdpkeyb"
[    29.909] (II) Loading /usr/lib/xorg/modules/input/xrdpkeyb_drv.so
[    29.910] (II) Module XRDPKEYB: vendor="X.Org Foundation"
[    29.910] 	compiled for 1.20.11, module version = 0.2.17
[    29.910] 	Module class: X.Org XInput Driver
[    29.910] 	ABI class: X.Org XInput driver, version 24.1
[    29.910] rdpkeybPlug:
[    29.910] rdpIdentify:
[    29.910] (II) XRDPDEV: driver for xrdp: XRDPDEV
[    29.911] rdpDriverFunc: op 10
[    29.911] (WW) Falling back to old probe method for XRDPDEV
[    29.911] rdpProbe:
[    29.911] (II) Loading sub module "fb"
[    29.911] (II) LoadModule: "fb"
[    29.911] (II) Loading /usr/lib/xorg/modules/libfb.so
[    29.911] (II) Module fb: vendor="X.Org Foundation"
[    29.911] 	compiled for 1.20.11, module version = 1.0.0
[    29.911] 	ABI class: X.Org ANSI C Emulation, version 0.4
[    29.911] rdpProbe: found DRMDevice xorg.conf value []
[    29.911] rdpProbe: found DRI3 xorg.conf value [1]
[    29.911] (II) XRDPDEV(0): using default device
[    29.911] (WW) VGA arbiter: cannot open kernel arbiter, no multi-card support
[    29.911] rdpPreInit:
[    29.911] rdpPreInit:  open failed
[    29.911] (**) XRDPDEV(0): Depth 24, (--) framebuffer bpp 32
[    29.911] (==) XRDPDEV(0): RGB weight 888
[    29.911] (==) XRDPDEV(0): Using gamma correction (1.0, 1.0, 1.0)
[    29.911] (==) XRDPDEV(0): Default visual is TrueColor
[    29.911] (==) XRDPDEV(0): DPI set to (96, 96)
[    29.911] (II) XRDPDEV(0): 	mode "640x480" ok
[    29.911] (II) XRDPDEV(0): 	mode "800x600" ok
[    29.911] (II) XRDPDEV(0): Virtual size is 800x600 (pitch 800)
[    29.911] (**) XRDPDEV(0):  Default mode "800x600": 36.0 MHz (scaled from 0.0 MHz), 35.2 kHz, 56.2 Hz
[    29.911] (II) XRDPDEV(0): Modeline "800x600"x0.0   36.00  800 824 896 1024  600 601 603 625 +hsync +vsync (35.2 kHz d)
[    29.912] rdpScreenInit: virtualX 800 virtualY 600 rgbBits 8 depth 24
[    29.912] rdpScreenInit: pfbMemory bytes 1920000
[    29.912] rdpScreenInit: pfbMemory 0x7f8711b010
[    29.912] rdpSimdInit: assigning yuv functions
[    29.912] (==) XRDPDEV(0): Backing store enabled
[    29.912] rdpClientConInit: disconnect idle session after [0] sec
[    29.912] rdpClientConInit: kill disconnected [0] timeout [0] sec
[    29.912] rdpXvInit: depth 24
[    29.912] rdpScreenInit: out
That's the log (just the interesting first half) after adding video/render groups back to user "pi", then doing the DRMDevice mod, rebooting and connecting again. The result is a slow/unstable/unusable remote connection.

**Comment by matt335672** [comment]
Log file looks fine.

Read [this post](/neutrinolabs/xrdp/issues/2060#issuecomment-981591912) regarding the slowness.

**Reply by reillysmithers** [reply]
Just tried DRMDevice solution and configured to use openbox instead of mutter, works much better! I'd call this the answer other than noticing some strange behavior such as white bezels around preview windows, and the inability to see a libcamera previews at all. It's nothing too concerning though so unless you know of some easy solution, I'm just gonna call it here. Thanks again!
